Creating static libraries for a multi platform project

Hello,
we are developing a mobile game and would like to switch to IXWebSocket for handling socket connections. For this we would need static libraries for the android (armeabi-v7a, arm64-v8a & x86), iOS and mac. We are struggling a bit to compile all these from the project and wondered if we could get some assistance for this case. 
The project states it is compatible with android, iOS and mac so would it be possible to get some more detailed instructions for building the project for these platforms? What where the commands/arguments used for compiling libs for these projects by you, the developers?
If all works out I would also be willing to post a PR which includes updates to the README pages for anyone who tries to do the same in the future.

So far I have been able to build a static lib for the mac target. For android I am using a customised version of the `build_android.sh` script.
```
${CMAKE} \
    [path]/IXWebSocket/ \
    -DANDROID_NATIVE_API_LEVEL=23 \
    -DANDROID_ABI=armeabi-v7a \
    -DANDROID_TOOLCHAIN=clang \
    -DANDROID_NDK=${ANDROID_NDK} \
    -G'Unix Makefiles' \
    -DCMAKE_TOOLCHAIN_FILE=${CMAKE_TOOLCHAIN_FILE} \
    -DCMAKE_MAKE_PROGRAM=make \
    -DOPENSSL_ROOT_DIR=/opt/homebrew/bin/openSSL \
    -DOPENSSL_LIBRARIES=/opt/homebrew/bin/openSSL/lib \
    -DUSE_TLS=1 \
    -DUSE_OPEN_SSL=1
```

But this results in:
```
-- TLS configured to use openssl
CMake Error at /Applications/CMake.app/Contents/share/cmake-3.26/Modules/FindPackageHandleStandardArgs.cmake:230 (message):
  Could NOT find OpenSSL, try to set the path to OpenSSL root folder in the
  system variable OPENSSL_ROOT_DIR (missing: OPENSSL_CRYPTO_LIBRARY
  OPENSSL_INCLUDE_DIR)
Call Stack (most recent call first):
  /Applications/CMake.app/Contents/share/cmake-3.26/Modules/FindPackageHandleStandardArgs.cmake:600 (_FPHSA_FAILURE_MESSAGE)
  /Applications/CMake.app/Contents/share/cmake-3.26/Modules/FindOpenSSL.cmake:670 (find_package_handle_standard_args)
  CMakeLists.txt:209 (find_package)
```

Any idea how to fix this?
Also I have yet to try building for iOS. As Ive never done this, I would also be grateful for help/pointers there.
Thank you in advance

Linking the android libraries seems to be clashing with our android NDK, resulting in the error `android-ndk/toolchains/llvm/prebuilt/darwin-x86_64/sysroot/usr/include/bits/fortify/poll.h:59: undefined reference to '__poll_chk'`. 
Is this simply an error indicating that our android NDK is outdated or would you know more about this?

--- a/docs/build.md
+++ b/docs/build.md
@@ -1,11 +1,37 @@
 ## Build
 
+### Building static libraries
+
+A static library can be build using:
+```
+mkdir build #make a build dir so that you can build out of tree
+cd build
+cmake -DUSE_TLS=1 ..
+make -j
+```
+There are also two build scripts in the `tools` folder for creating static libraries for android and iOS/macOS respectively.<br>
+Arguments can be changed in the script files.
+
+Android:
+```
+mkdir build
+cd build
+./../tools/build_android.sh
+make -j
+```
+macOS & iOS:
+```
+mkdir build
+cd build
+./../tools/build_ios.sh
+```
+
 ### CMake
 
 CMakefiles for the library and the examples are available. This library has few dependencies, so it is possible to just add the source files into your project. Otherwise the usual way will suffice.
 
 ```
-mkdir build # make a build dir so that you can build out of tree.
+mkdir build
 cd build
 cmake -DUSE_TLS=1 ..
 make -j

--- a/tools/build_android.sh
+++ b/tools/build_android.sh
@@ -15,7 +15,13 @@
 # -DANDROID_NATIVE_API_LEVEL=23
 # -DANDROID_TOOLCHAIN=clang
 # jvmArgs :
-# 
+#
+# To use a specific version of openSSL, add the arguments:
+# -DOPENSSL_VERSION="[version]"
+# -DOPENSSL_FOUND=1
+# -DOPENSSL_INCLUDE_DIR=[include path]
+# -DOPENSSL_LIBRARIES="[lib path]/libssl.a;[lib path]/libcrypto.a"
+#
 
 CMAKE_TOOLCHAIN_FILE=/tools/android/android-ndk-r20-darwin/build/cmake/android.toolchain.cmake
 ANDROID_HOME=/tools/android/android-sdk-darwin
@@ -33,4 +39,4 @@ ${CMAKE} \
     -G'Unix Makefiles' \
     -DCMAKE_TOOLCHAIN_FILE=${CMAKE_TOOLCHAIN_FILE} \
     -DCMAKE_MAKE_PROGRAM=make \
-    -DUSE_WS=1
+    -DUSE_TLS=1

--- a/tools/build_ios.sh
+++ b/tools/build_ios.sh
@@ -1,0 +1,35 @@
+#!/bin/sh
+
+#
+# This script creates static libraries for macOS and iOS devices
+#
+# To use a specific OpenSSL version add the arguments:
+# -DOPENSSL_VERSION="[version]" \
+# -DOPENSSL_FOUND=1 \
+# -DOPENSSL_INCLUDE_DIR=[include path] \
+# -DOPENSSL_LIBRARIES="[lib path]/libssl.a;[lib path]/libcrypto.a" \
+#
+
+DEPLOYMENT_TARGET_IOS='12.0'
+DEPLOYMENT_TARGET_MAC='14.0'
+CMAKE_DIR=/Applications/CMake.app/Contents/bin
+CMAKE=${CMAKE_DIR}/cmake
+
+${CMAKE} \
+    .. \
+    -GXcode \
+    -DUSE_TLS=1
+
+mkdir -p ios sim mac
+
+xcodebuild -project 'ixwebsocket.xcodeproj' -target "ixwebsocket" -sdk iphoneos ARCHS='arm64' TARGET_BUILD_DIR='$(PWD)/ios' BUILT_PRODUCTS_DIR='$(PWD)/ios' TARGET_NAME='ixwebsocket' IPHONEOS_DEPLOYMENT_TARGET=${DEPLOYMENT_TARGET_IOS}
+xcodebuild -project 'ixwebsocket.xcodeproj' -target "ixwebsocket" -sdk iphonesimulator ARCHS='x86_64' TARGET_BUILD_DIR='$(PWD)/sim' BUILT_PRODUCTS_DIR='$(PWD)/sim' TARGET_NAME='ixwebsocket' IPHONEOS_DEPLOYMENT_TARGET=${DEPLOYMENT_TARGET_IOS}
+xcodebuild -project 'ixwebsocket.xcodeproj' -target "ixwebsocket" -destination 'platform=OS X' ARCHS='x86_64 arm64' TARGET_BUILD_DIR='$(PWD)/mac' BUILT_PRODUCTS_DIR='$(PWD)/mac' TARGET_NAME='ixwebsocket' MACOSX_DEPLOYMENT_TARGET=${DEPLOYMENT_TARGET_MAC}
+
+rm -rf ./out/*.a
+rm -rf ./out/ios/*.a
+rm -rf ./out/mac/*.a
+mkdir -p out
+mkdir -p ./out/ios ./out/mac
+lipo -create "./ios/libixwebsocket.a" "./sim/libixwebsocket.a" -output ./out/ios/libixwebsocket.a
+cp "./mac/libixwebsocket.a" ./out/mac/libixwebsocket.a

